fwprovider/cluster/metrics/metrics_server_model.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package metrics
+
+import (
+	"github.com/bpg/terraform-provider-proxmox/proxmox/cluster/metrics"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type metricsServerModel struct {
+	ID                  types.String `tfsdk:"id"`
+	Name                types.String `tfsdk:"name"`
+	Disable             types.Bool   `tfsdk:"disable"`
+	MTU                 types.Int64  `tfsdk:"mtu"`
+	Port                types.Int64  `tfsdk:"port"`
+	Server              types.String `tfsdk:"server"`
+	Timeout             types.Int64  `tfsdk:"timeout"`
+	Type                types.String `tfsdk:"type"`
+	InfluxAPIPathPrefix types.String `tfsdk:"influx_api_path_prefix"`
+	InfluxBucket        types.String `tfsdk:"influx_bucket"`
+	InfluxDBProto       types.String `tfsdk:"influx_db_proto"`
+	InfluxMaxBodySize   types.Int64  `tfsdk:"influx_max_body_size"`
+	InfluxOrganization  types.String `tfsdk:"influx_organization"`
+	InfluxToken         types.String `tfsdk:"influx_token"`
+	InfluxVerify        types.Bool   `tfsdk:"influx_verify"`
+	GraphitePath        types.String `tfsdk:"graphite_path"`
+	GraphiteProto       types.String `tfsdk:"graphite_proto"`
+}
+
+// importFromAPI takes data from metrics server PVE API response and set fields based on it.
+// Note: API response does not contain name so it must be passed directly.
+func (m *metricsServerModel) importFromAPI(name string, data *metrics.ServerData) {
+	m.ID = types.StringValue(name)
+	m.Name = types.StringValue(name)
+
+	var disable *bool
+
+	if data.Disable != nil {
+		if *data.Disable == 1 {
+			*disable = true
+		} else {
+			*disable = false
+		}
+	}
+
+	m.Disable = types.BoolPointerValue(disable)
+
+	m.MTU = types.Int64PointerValue(data.MTU)
+	m.Port = types.Int64PointerValue(data.Port)
+	m.Server = types.StringPointerValue(data.Server)
+	m.Timeout = types.Int64PointerValue(data.Timeout)
+	m.Type = types.StringPointerValue(data.Type)
+	m.InfluxAPIPathPrefix = types.StringPointerValue(data.APIPathPrefix)
+	m.InfluxBucket = types.StringPointerValue(data.Bucket)
+	m.InfluxDBProto = types.StringPointerValue(data.InfluxDBProto)
+	m.InfluxMaxBodySize = types.Int64PointerValue(data.MaxBodySize)
+	m.InfluxOrganization = types.StringPointerValue(data.Organization)
+	m.InfluxToken = types.StringPointerValue(data.Token)
+
+	var influxVerify *bool
+
+	if data.Verify != nil {
+		if *data.Verify == 1 {
+			*influxVerify = true
+		} else {
+			*influxVerify = false
+		}
+	}
+
+	m.InfluxVerify = types.BoolPointerValue(influxVerify)
```

Comment:
I believe it can be simplified like this:
```suggestion
	if data.Verify != nil {
		m.InfluxVerify = types.BoolValue(*data.Verify == 1)
	}
```

and same for `metricsServerDatasourceModel` below

Reply:
sorry, this is not the same and i want to be precise here

for proxmox, "nothing" aka `nil` in this code, means "use PVE default" where `types.BoolValue(*data.Verify == 1)` is always either true or false (that will be send to proxmox and saved in state like that)

But you are right that this was ugly 

I added **boolToInt64Ptr**, **int64ToBoolPtr** on top of this file to keep those details away and have it in one place. maybe later one there if there are more use cases, we can make new custom type for (*bool <-> *int64), hope it's fine

Reply:
ah, you're absolutely right about nulls here!

there is actually `proxmox/types.CustomBool` that can handle this use case.

i'm ok to leave `boolToInt64Ptr` and `int64ToBoolPtr` functions for now, and refactor the code later.